### PR TITLE
Update getLastValue docs

### PR DESCRIPTION
**Pull request by aljazfrancic**
## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] Pull request title reflects its content
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# Description:

### CODE REVIEW

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 1]
```diff
@@ -1,33 +1,374 @@
...
 Python::
 +
 [source,python]
 ----
-def HowToDoc_MeasureSingleValue(device):
-    signal = device.signals_recursive[0]
-    measurementValue = signal.last_value
-    print ("Single measurement value: [",measurementValue,"]")
+# Get last value of a Data Packet
+packet.last_value()  # TODO ()???
```
Comment (aljazfrancic) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:49]: Not sure if brackets are appropriate here... It's suppored to take one argument for Struct and the interpreter suggested it. Couldn't test Python code...

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 2]
```diff
@@ -1,33 +1,374 @@
...
+descriptor_int = builder_int.build()
+
+builder_float = opendaq.DataDescriptorBuilder()
+builder_float.name = "MyFloat64"
+builder_float.sample_type = opendaq.SampleType.Float64
+descriptor_float = builder_float.build()
+
+builder = opendaq.DataDescriptorBuilder()
+builder.name = "MyStruct"
+builder.sample_type = opendaq.SampleType.Struct  # TODO not implemented
```
Comment (aljazfrancic) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:275]: SampleType.Struct is not supported in Python yet.

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 3]
```diff
@@ -1,33 +1,374 @@
...
+struct_fields = opendaq.List()
+struct_fields.append(descriptor_int)
+struct_fields.append(descriptor_float)
+builder.struct_fields = struct_fields
+descriptor = builder.build()
+
+# Set the Data Descriptor, thereby adding MyStruct to the Type Manager
+signal.descriptor = descriptor  # TODO setter not defined
```
Comment (aljazfrancic) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:283]: This setter doesn't seem to be defined yet.

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc (hunk 1 above)
Comment (aljazfrancic) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:49]: Not sure if brackets are appropriate here... It's supported to take one argument for Struct and the interpreter suggested it. Couldn't test Python code (however I have some snippets in a separate branch)...

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 4]
```diff
@@ -1,33 +1,374 @@
...
+----
+# Print last value
+print("last value: ", last_value)
+----
+====
+
+[#casting_the_last_value]
+== Casting the last value and using the cast variable
+
+In C++, if we would like to do something useful with the `IBaseObject`, such as access its underlying methods, we need to cast it into the correct type first. This is not necessary in Python, however.
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:79]:
```suggestion
In C++, the function `getLastValue` returns an object of type IBaseObject. To manipulate it, developers must explicitly convert it to the appropriate type. However, in Python, this conversion process is handled automatically.
```

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 5]
```diff
@@ -1,33 +1,374 @@
...
+
+The underlying type of the `IBaseObject` we get from our `getLastValue` call can be cast into either `IFloat` if the type is `Float32` or `Float64`, to `IInteger` if the type is `Int8` through `Int64` or `UInt8` through `UInt64`, to `IComplexNumber` if type is `ComplexFloat32` or `ComplexFloat64`, to `IRange` if the type is `RangeInt64`, to `IStruct` if the type is `Struct`, and to `IList` of the aforementioned types if there is exactly one dimension in the single value.
+
+In this example, we cast `lastValue` object into an `IFloat` like so:
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:83]:
```suggestion
example of casting the `lastValue` object into an `IFloat` object:
```

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 6]
```diff
@@ -1,33 +1,374 @@
...
+Cpp::
++
+[source,cpp]
+----
+// Cast to IFloat
+auto number = lastValue.asPtr<IFloat>();
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which we call `getLastValue` (`last_value` in Python) is:
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:96]: remove `we`

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 7]
```diff
@@ -1,33 +1,374 @@
...
+builder = opendaq.DataDescriptorBuilder()
+builder.sample_type = opendaq.SampleType.Float64
+descriptor = builder.build()
+----
+====
+
+[#last_value_range]
+== Getting the last value with a Range
+
+In the following example, we cast the `lastValue` object into an `IRange` like so:
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:121]:
```suggestion
The following example demonstrate casting the `lastValue` object into an `IRange`
```

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 8]
```diff
@@ -1,33 +1,374 @@
...
+[source,cpp]
+----
+// Cast to IRange
+auto range = lastValue.asPtr<IRange>();
+----
+====
+
+An example of something we could now do is call the methods of the underlying type and access its members such as the `getLowValue` (`low_value` in Python) and the `getHighValue` (`high_value` in Python), in this case provided by the `IRange` interface.
+
+[tabs]
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:136]: i would squash this example with previous one

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 9]
```diff
@@ -1,33 +1,374 @@
...
++
+[source,python]
+----
+# Extract values
+low = last_value.low_value
+high = last_value.high_value
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which we call `getLastValue` (`last_value` in Python) is:
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:156]: remove 'we'

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 10]
```diff
@@ -1,33 +1,374 @@
...
+
+[tabs]
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:158]: i would remove this example and add comment that we are assuming that type of data descriptor is int64

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 11]
```diff
@@ -1,33 +1,374 @@
...
+builder = opendaq.DataDescriptorBuilder()
+builder.sample_type = opendaq.SampleType.RangeInt64
+descriptor = builder.build()
+----
+====
+
+[#last_value_complex_number]
+== Getting the last value with a Complex Number
+
+In the following example, we first extract the last value via `getLastValue` (`last_value` in Python) call, then we cast the `lastValue` object into a `IComplexNumber` type (not needed in Python). Lastly, call the methods of the underlying type such as the `getReal` (`real` in Python) and the `getImaginary` (`imaginary` in Python), in this case provided by the `IComplexNumber` interface.
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:181]: - remove `we`
- you can make a list of steps (up ur choice )

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 12]
```diff
@@ -1,33 +1,374 @@
...
+[source,python]
+----
+# Get last value of a Signal
+last_value = signal.last_value
+# Extract values
+real = last_value.real
+imaginary = last_value.imaginary
+====
+
+A Data Descriptor for a Complex Number might be:
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:208]: just add assumption in prev exmaple. and remove this example

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc (hunk 2 above)
Comment (denise-opendaq) [reply docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:275]: add note in doc

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc (hunk 3 above)
Comment (denise-opendaq) [reply docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:283]: setter in ISignalConfig

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 13]
```diff
@@ -1,33 +1,374 @@
...
+struct_fields.append(descriptor_float)
+builder.struct_fields = struct_fields
+descriptor = builder.build()
+
+# Set the Data Descriptor, thereby adding MyStruct to the Type Manager
+signal.descriptor = descriptor  # TODO setter not defined
+----
+====
+
+In the above example, we first build a Data Descriptor of a custom Struct that has two fields. The first field has Sample Type `Int32` and the second one has Sample Type `Float64`.
```
Comment (denise-opendaq) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:287]: remove 'we'

**Review by denise-opendaq — CHANGES_REQUESTED** [review]

**Review by denise-opendaq — APPROVED** [review]

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 14]
```diff
@@ -1,33 +1,305 @@
...
+Python::
++
+[source,python]
+----
+
+----
+====
+
+[#last_value_range]
+== Getting the last value with a Range
```
Comment (JakaMohorkoDS) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:126]: I'd omit this section. We just need a list of type conversions at the top for users to refer to. Beyond that they should understand how to use certain openDAQ types - this should be a separate guide if we need it.

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 15]
```diff
@@ -1,33 +1,305 @@
...
-    signal = device.signals_recursive[0]
-    measurementValue = signal.last_value
-    print ("Single measurement value: [",measurementValue,"]")
+
 ----
 ====
 
+[CAUTION]
+====
+If we want to call `getLastValue` (`last_value` in Python) on a Data Packet (as opposed to a Signal) and the underlying data type is a Struct, we must provide the optional parameter, Type Manager, which has had the custom Struct added to it.
```
Comment (JakaMohorkoDS) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:263]: Honestly, I would not even mention that we can read the last value on a packet. The guide aims to describe how to read the last value of a signal, the packet last value is mostly for internal usage and will confuse readers.

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 16]
```diff
@@ -1,33 +1,305 @@
...
+[source,python]
+----
+
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which `getLastValue` (`last_value` in Python) is called is similar to the the one from the previous example (the Data Descriptor with Sample Type `Float64`), except the Sample Type is `RangeInt64`.
+
+[#last_value_complex_number]
+== Getting the last value with a Complex Number
```
Comment (JakaMohorkoDS) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:153]: As above, I don't think an explanation of how complex numbers work is needed in this how-to

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 17]
```diff
@@ -1,33 +1,305 @@
...
+----
+
+====
+
+A Data Descriptor for this Complex Number example is again very similar to the ones from previous examples, except the Sample Type might be `ComplexFloat32` or `ComplexFloat64` or  instead of `Float64` or `RangeInt64`.
+
+[#last_value_struct]
+== Getting the last value with a Struct
+
+A Data Descriptor for a Struct might be:
```
Comment (JakaMohorkoDS) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:187]: We should simply describe the descriptor and how the signal in question is structured. No need to provide the builder code sample.

File: docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc [hunk 18]
```diff
@@ -1,33 +1,150 @@
...
+----
+# Get last value of a Signal
+my_last_value = my_signal.last_value
+----
+====
+
+[#last_value_signal_types]
+== Last value and Signal types
+
+The method for getting the last value yields an openDAQ Base object-type value.
```
Comment (JakaMohorkoDS) [inline docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc:33]: A 1-line code example of how to retrieve the value type would be great here:

```
SampleType type = signal.getDescriptor().getSampleType();
```

**Review by JakaMohorkoDS — APPROVED** [review]

**Review by JakaMohorkoDS — APPROVED** [review]

**Pull request merged**